src/main/scala/gameengine/GameEngine.scala

```diff
@@ -0,0 +1,143 @@
+package gameengine
+
+import characters.{GameBoard, Person, Player, Resources}
+
+import scala.Console.println
+import scala.collection.mutable.ListBuffer
+import scala.io.StdIn.readLine
+import scala.util.Random
+
+class GameEngine {
+  val resources = new Resources
+  private var questionList: List[String] = resources.listOfQuestions
+  private val random = new Random()
+  // Instantiate the game board
+  val gameBoard = new GameBoard(resources.charactersList)
+  // create firstPlayer
+  private val firstPlayer: Player = createPlayerAndAssignGameBoard()
+  private var continuedPlaying: Boolean = true
+
+  /**
+   * Create an object of player by assigning name, gameBoard and secret characters
+   * @return player object
+   */
+  def createPlayerAndAssignGameBoard(): Player = {
+    val player = new Player(name = "Player1", gameBoard = gameBoard.gameBoardForPlayer,
+      selectRandomCharacter(resources.charactersList))
+    player
+  }
+
+  /**
+   * Starts the Game by calling all the necessary methods
+   */
+  def startTheGame(): String = {
+    while (continuedPlaying) {
+      if (endGame(firstPlayer.gameBoard)) {
+        println("Congratulations, you won!")
```

Comment:
Spoke too soon can see you have used endGame. Consider making a branch for extension off your MVP branch or when making PR for extension if you are not going to merge your PR's make the PR so that rather than comparing to main you compare to mvp branch so this PR is only the differences between the 2. e.g. https://github.com/BAmiriDev/GuesswhoGame/compare/MVP...Extension1